### PR TITLE
Add architecture DFD files

### LINKED ISSUE #21: Create architecture DFD
In order to better understand system structure, it is essential to have a DFD diagram as a visual representation of the information flows within the system.

### PR DESCRIPTION
**Pull request by Ofekmn**
resolves #21

### CODE REVIEW

**Review by rimonsh — APPROVED** [review]

**Review by Sendy459 — DISMISSED** [review]
Looks good to me

**Review by rotem2022 — APPROVED** [review]

**Review by rotem2022 — APPROVED** [review]

**Review by ormergi — CHANGES_REQUESTED** [review]
Hi, Ofek, please make sure to squash intermediate commits at least when the discussion is done and you got approve from a maintainer.

I think the diagram lack context in the sense of who performs the operation (the person that created the poll or the is it participants) and where it starts (chronologically).

Could you please add an entity that reflects the end user in each flow?
Also, please make sure the flow is described in a chronological way (e.g: right to left) where possible.

**Review by rotem2022 — DISMISSED** [review]

**Comment by Ofekmn** [comment]
> Hi, Ofek, please make sure to squash intermediate commits at least when the discussion is done and you got approve from a maintainer.
> 
> I think the diagram lack context in the sense of who performs the operation (the person that created the poll or the is it participants) and where it starts (chronologically).
> 
> Could you please add an entity that reflects the end user in each flow? Also, please make sure the flow is described in a chronological way (e.g: right to left) where possible.

The chronological order is from up to down when possible.

**Review by ormergi — APPROVED** [review]

**Review by Sendy459 — DISMISSED** [review]

File: uml/DFD_diagram.png
Comment (kobihk) [inline uml/DFD_diagram.png:1]: @Ofekmn I can see that most of the Entities on the left side are User
maybe we can create one Entity for all

**Review by kobihk — CHANGES_REQUESTED** [review]

**Review by Sendy459 — DISMISSED** [review]

**Review by kobihk — CHANGES_REQUESTED** [review]

**Review by kobihk — APPROVED** [review]

**Review by Sendy459 — APPROVED** [review]

**Review by rotem2022 — APPROVED** [review]

**Pull request merged**